src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -676,6 +667,42 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+#nullable enable
+        private async Task<IReadOnlyList<PackageSource>> GetAllPackageSourcesAsync()
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var remoteBroker = await BrokeredServicesUtilities.GetRemoteServiceBrokerAsync();
+            var nugetSourcesService = await remoteBroker.GetProxyAsync<INuGetSourcesService>(NuGetBrokeredServices.SourceProviderService);
+
+            if (nugetSourcesService != null)
+            {
+                using (nugetSourcesService)
+                {
```

Comment:
Since the order of these two blocks tend to need to be inverted to satisfy the analyzers, I suspect you're not running them.
How are you getting the `Microsoft.ServiceHub.Framework` assembly reference? Can you make sure you have a `PackageReference` to the latest version so the analyzers can help keep you on the right track?

Reply:
Sure enough, we were out of date on our VS references, updating allowed me to discover the Async methods for DialogPage and fix this as well.